Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs

```diff
@@ -95,6 +95,38 @@ private void TrackEvent_Click(object sender, RoutedEventArgs e)
             Analytics.TrackEvent(name, propertiesDictionary);
         }
 
+        private void CountryCodeEnabled_Checked(object sender, RoutedEventArgs e)
+        {
+            if (cbEnableCountryCode.IsChecked.HasValue)
+            {
+                if (!cbEnableCountryCode.IsChecked.Value)
+                {
+                    lCountryCode.Text = "";
+                    AppCenter.SetCountryCode(null);
+                }
+                else
+                {
+                    lCountryCode.Text = RegionInfo.CurrentRegion.TwoLetterISORegionName;
+                    AppCenter.SetCountryCode(lCountryCode.Text);
+                }
+                spCountryCode.IsEnabled = cbEnableCountryCode.IsChecked.Value;
+            }
+        }
+
+        private void CountryCodeSave_ClickListener(object sender, RoutedEventArgs e)
+        {
+            // Checked that input country code is valid.
+            bool isValidCountryCode = CultureInfo.GetCultures(CultureTypes.SpecificCultures)
```

Comment:
Why do we test if the code is valid? Might be a valid test scenario to set an invalid country code.